pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -449,6 +487,12 @@ const (
 	GracefullyStoppedSkip SkippingReason = "PipelineRun was gracefully stopped"
 	// MissingResultsSkip means the task was skipped because it's missing necessary results
 	MissingResultsSkip SkippingReason = "Results were missing"
+	// PipelineTimedOutSkip means the task was skipped because the PipelineRun has passed its overall timeout.
+	PipelineTimedOutSkip SkippingReason = "PipelineRun timeout has been reached"
```

Comment:
what is the existing behavior in terms what happens to the unscheduled tasks (dag and finally)? How is this different from what we have today?

Reply:
Right now, we create the tasks with a 1s timeout, so that they get created but immediately timeout. Skipping them instead is much more logical, imo.

Reply:
Wouldn't that be considered a behavior change? The users will start getting such tasks as part of the list of skipped tasks instead of `taskRun`s with cancelled status? 🤔  

> Skipping them instead is much more logical, imo.

`skip` was introduced for conditional tasks and since then we have extended its usage in many different scenarios which is fine for now.

Reply:
It is a slight behavior change, but not a functional one, since the end result is the same, just without the wasted steps of creating and immediately timing out the relevant task(s). I don’t think it’s an issue.

Reply:
Agreeing with @abayer